### ISSUE TITLE
Update the panel splitter to follow the Window Splitter ARIA pattern

### ISSUE THREAD

**Issue by barklund** [GoogleForCreators/web-stories-wp#3856]
## Feature Description

We have a [panel handle](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/panels/panel/shared/handle.js) currently with a custom implementation for interaction behaviour. This should be updated to follow [the Window Splitter specification](https://www.w3.org/TR/wai-aria-practices-1.1/#windowsplitter).

## Additional Context

* Add `role="separator"`
* Add correct `aria-*` attributes for min, max and value
* Add `aria-labelledby` with reference to panel title, if exists (maybe require it?)
* Add `aria-controls` with reference to entire panel
* Make <kbd>Enter</kbd> collapse/expand the panel
* Update tests (add test for enter, use role when selecting)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by swissspidy** [comment]
Is this still relevant?
